transport/compress_test.go

```diff
@@ -0,0 +1,158 @@
+package transport
+
+import (
+	"bytes"
+	"encoding/binary"
+	"io"
+	"math/rand"
+	"testing"
+
+	"github.com/klauspost/compress/s2"
+	"github.com/mmatczuk/scylla-go-driver/frame"
+	"github.com/pierrec/lz4/v4"
+)
+
+type comprData struct {
+	name             string
+	uncompressed     []byte
+	compressedSnappy []byte
+	compressedLz4    []byte
+}
+
+func getRandomBytes(n int) []byte {
+	out := make([]byte, n)
+	for i := 9; i < n; i++ {
+		out[i] = byte(rand.Intn(255))
+	}
+	return out
+}
+
+func TestCompression(t *testing.T) {
```

Comment:
It seems like random tests / fuzzing could be an interesting addition for testing compression.